apps/prairielearn/assets/scripts/instructorAssessmentAccessClient.ts

```diff
@@ -0,0 +1,216 @@
+      .toString()
+      .split(',')
+      .map((uid: string) => uid.trim());
+    updatedAccessRules.active = updatedAccessRules.active === 'true';
+
+    updatedAccessRules.start_date = new Date(
+      Temporal.PlainDateTime.from(updatedAccessRules.start_date)
+        .toZonedDateTime(timezone)
+        .toInstant().epochMilliseconds,
+    ).toISOString();
+
+    updatedAccessRules.end_date = new Date(
+      Temporal.PlainDateTime.from(updatedAccessRules.end_date)
+        .toZonedDateTime(timezone)
+        .toInstant().epochMilliseconds,
+    ).toISOString();
+
+    if (accessRulesData[row]) {
+      accessRulesData[row].assessment_access_rule = updatedAccessRules;
+    } else {
+      accessRulesData[row] = { assessment_access_rule: updatedAccessRules };
+    }
+    refreshTable();
+  }
+
+  on('click', '.editButton', (e) => {
+    const editButton = (e.target as HTMLElement).closest('button');
+    if (!editButton) return;
+
+    templateFromAttributes(editButton, editAccessRuleModal as HTMLElement, {
+      'data-row': '.access-rule-row',
+      'data-title-text': '.modal-title',
+      'data-submit-text': '.updateAccessRuleButton',
+      'data-access-rule-mode': '.access-rule-mode',
+      'data-access-rule-uids': '.access-rule-uids',
+      'data-access-rule-start-date': '.access-rule-start-date',
+      'data-access-rule-end-date': '.access-rule-end-date',
+      'data-access-rule-active': '.access-rule-active',
+      'data-access-rule-credit': '.access-rule-credit',
+      'data-access-rule-time-limit': '.access-rule-time-limit',
+      'data-access-rule-password': '.access-rule-password',
+      'data-access-rule-exam-uuid': '.access-rule-exam-uuid',
+    });
+
+    $('#editAccessRuleModal').modal('show');
+  });
+
+  on('click', '#updateAccessRuleButton', (e) => {
+    const form = (e.target as HTMLElement).closest('form');
+    if (!form) return;
+    handleUpdateAccessRule({ form });
+  });
+
+  function swapRows(row: any, targetRow: any) {
+    const accessRule = accessRulesData[row];
+    accessRulesData[row] = accessRulesData[targetRow];
+    accessRulesData[targetRow] = accessRule;
+    refreshTable();
+  }
+
+  on('click', '.up-arrow-button', (e) => {
+    const row = parseInt((e.target as HTMLElement).closest('button')?.dataset.row ?? '0');
+    if (!row) return;
+    swapRows(row, row - 1);
+  });
+
+  on('click', '.down-arrow-button', (e) => {
+    const row = parseInt((e.target as HTMLElement).closest('button')?.dataset.row ?? '0');
+    if (row === accessRulesData.length - 1) return;
+    swapRows(row, row + 1);
+  });
+
+  on('click', '#addRuleButton', () => {
+    const addRuleButton = document.getElementById('addRuleButton');
+    if (!addRuleButton) return;
+
+    templateFromAttributes(addRuleButton, editAccessRuleModal as HTMLElement, {
+      'data-row': '.access-rule-row',
+      'data-title-text': '.modal-title',
+      'data-submit-text': '.updateAccessRuleButton',
```

Comment:
This feels like another case where just calling `EditAccessRuleModal` with an explicit title and such would be better.

Reply:
Updated to now call `EditAccessRuleModal` when needed in [660f81b](https://github.com/PrairieLearn/PrairieLearn/pull/9934/commits/660f81bebe37d51fcef94b1535961f4abfa02d88).